File: cats/oaip/semester 1/F1 Elite number/solve.py
def correct(num, index):
    num = list(str(num))
    f = "".join(num[0:index])
    s = str(int(num[index]) + 1)
    t = len(num[index + 1:])
    t *= "1"
    return f + s + t


def is_elite(num):
    sub = int(num)
    while sub > 0:
        last = sub % 10
        if last == 0:
            return False
        if num % last != 0:
            return False
        sub //= 10
    return True

def skip(num):
    number = list(str(num))
    flag_even = False
    flag_zero = False
    flag_five = False
    for i in range(len(number)):
        if number[i] == "0":
            flag_zero = True
        if number[i] in ("2", "4", "6", "8"):
            flag_even = True
        if number[i] == "5":
            flag_five = True
        if (flag_five and flag_even) or flag_zero:
            num = int(correct(num, i))
            break
    return num

with open("input.txt", "r") as inputfile, open("output.txt", "w") as out:
    n = int(inputfile.readlines()[0])
    while not is_elite(n):
        n = skip(n + 1)

    out.write(str(n))

File: cats/oaip/semester 2/E1 Блинчики/solve.cc
#include <fstream>

using namespace std;

int main() {

    ifstream input_file("input.txt");
    ofstream output_file("output.txt");

    int n, res = 0;
    char a, b;

    input_file >> n;
    input_file >> a;

    for (int i = 1; i < n; ++i) {

        input_file >> b;
        if (a != b) {
            res++;
            a = b;
        }
    }
    if (b == 'B') res++;

    output_file << res;
}